sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -115,8 +116,20 @@ def authorize_request(self, request: PipelineRequest[HTTPRequestType], *scopes:
         """
         if self._enable_cae:
             kwargs.setdefault("enable_cae", self._enable_cae)
-        self._token = self._credential.get_token(*scopes, **kwargs)
-        self._update_headers(request.http_request.headers, self._token.token)
+
+        if hasattr(self._credential, "get_token_info"):
```

Comment:
How about having a helper function to handle 

if
call get_token_info
else
call get_token

in a central place?

Reply:
The point is get_token does not support pop.

Later we may need to add the logic

if pop
only try get_token_info

If we can put it in a central place, it can increase the maintainability.

Reply:
Sure, I centralized the logic to a private method in the respective policy classes.